### PR TITLE
fix(modal): ensure correct index is set for new modals

### LINKED ISSUE #5670: multiple modals & z-index issue
### Bug description:
Let's say multiple modals are opened at the same time - modals [A, B, C, D], opened in this order. If I close the first opened modal A, then open another modal E -> [B, C, D, E]. The newly opened modal E has the same index & z-index as the previous modal D. With the current reversed order of modals in DOM & same z-index, modal E will not be visible (will be covered by modal D).

### Link to minimally-working plunker that reproduces the issue:
[https://plnkr.co/edit/GNZtzlaxDM0X8STp8fTD](https://plnkr.co/edit/GNZtzlaxDM0X8STp8fTD)
In the plunkr, i set the limit to 5. So the issue is obvious after 5 opened modals...

### Version of Angular, UIBS, and Bootstrap
Angular 1.4.9
UIBS 1.1.2
Bootstrap 3.3.6

### PR DESCRIPTION
**Pull request by josetaira**
Fix how the index for the modal and the backdrop are set, depending on
what index value the previous top modal has.

Closes #5670

### CODE REVIEW

File: src/modal/modal.js [hunk 1]
```diff
@@ -278,6 +280,12 @@ angular.module('ui.bootstrap.modal', ['ui.bootstrap.stackedMap', 'ui.bootstrap.p
             topBackdropIndex = i;
           }
         }
+
+        // If any backdrop exist, ensure that it's index is always
+        // right below the top modal
+        if(topBackdropIndex > -1 && topBackdropIndex < topModalIndex) {
```
Comment (wesleycho) [inline src/modal/modal.js]: Add a space after `if`

File: src/modal/modal.js [hunk 2]
```diff
@@ -293,6 +301,10 @@ angular.module('ui.bootstrap.modal', ['ui.bootstrap.stackedMap', 'ui.bootstrap.p
 
         //clean up the stack
         openedWindows.remove(modalInstance);
+        previousTopOpenedModal = openedWindows.top();
+        if(previousTopOpenedModal) {
```
Comment (wesleycho) [inline src/modal/modal.js]: Ditto

File: src/modal/modal.js [hunk 3]
```diff
@@ -293,6 +301,10 @@ angular.module('ui.bootstrap.modal', ['ui.bootstrap.stackedMap', 'ui.bootstrap.p
...
+          topModalIndex = parseInt(previousTopOpenedModal.value.modalDomEl.attr("index"), 10);
```
Comment (wesleycho) [inline src/modal/modal.js]: Use single quotes.

**Comment by wesleycho** [comment]
Can you add a unit test for this?

**Comment by josetaira** [comment]
For the tests, no idea how grunt works, but I guess I can learn it. Do you have any templates or guidelines on how tests should be made?

**Comment by icfantv** [comment]
@josetaira, you don't need to learn grunt, you need to look at the `model.spec.js` file and add your tests to that file and they'll run as part of the build.  there should be enough examples in there (and our other test files) to figure it out.

if you want to run a specific test or test suite, simply prefix the `it` or `describe` with an `f` (i.e. `fit` or `fdescribe`) and only that test will be run.  our linter will reject this as invalid syntax so you can run `karma start` to run the karma tests independently.

**Comment by josetaira** [comment]
@icfantv Thanks for the info. Working on the unit tests now. Seems I did have to use grunt to create some templates (run `grunt default` before any tests). I'll add a commit for the test later on.

**Comment by icfantv** [comment]
@josetaira good deal.

**Comment by josetaira** [comment]
@icfantv @wesleycho tests committed.

**Comment by wesleycho** [comment]
This LGTM - thanks for the effort!

**Pull request closed without merging**